### PR TITLE
Security fixes

### PR DESCRIPTION
**Pull request by ggalmazor**
This PR adds extra security protections to some RPC calls:
- Change users from Site Admin > Edit Users
- Delete a form
- Delete submission
- Purge published data
- Purge submissions

#### What has been done to verify that this works as intended?
First, I've manually verified that all these operations still work in my local Aggregate instance
Second, I've inspected the network interchange of the delete form action and I've resent it using curl from the command line to verify that it won't work since the CSRF token wasn't valid anymore.

#### Why is this the best possible solution? Were any other approaches considered?
This solution implements the [official instructions](http://www.gwtproject.org/doc/latest/DevGuideSecurityRpcXsrf.html) to add CSRF protection to GWT apps.

To reduce code duplication and complexity, I've created a wrapper that handles client-side CSRF requests.

#### Are there any risks to merging this code? If so, what are they?
Any third party hitting these RPCs will have to request a CSRF token before making them. 

#### Do we need any specific form for testing your changes? If so, please attach one
No.

#### Does this change require updates to documentation? If so, please file an issue at https://github.com/opendatakit/docs/issues/new and include the link below.
No.

### CODE REVIEW

**Comment by dcbriccetti** [comment]
I expect to look at his today.

**Comment by dcbriccetti** [comment]
So far this looks good, but as someone who is only marginally able to *run* Aggregate (and even then with tons of errors on startup that don’t seem to break the features I need for running Briefcase), I can’t vet this change with confidence in my ability to understand it.

**Comment by ggalmazor** [comment]
Thanks, @dcbriccetti!

File: src/main/java/org/opendatakit/aggregate/client/permissions/AccessConfigurationSheet.java [hunk 1]
```diff
@@ -82,7 +80,9 @@
     userType = new ArrayList<String>();
     userType.add(ACCOUNT_TYPE_ODK);
     userType.add(ACCOUNT_TYPE_GOOGLE);
-  };
+  }
+
+  ;
```
Comment (dcbriccetti) [inline src/main/java/org/opendatakit/aggregate/client/permissions/AccessConfigurationSheet.java]: By reformatting and exposing these unnecessary semicolons—I count ten—you take on the responsibility of deleting them. 😀

Comment (ggalmazor) [reply src/main/java/org/opendatakit/aggregate/client/permissions/AccessConfigurationSheet.java]: I wonder how on earth did I miss those! Removing them :)

**Comment by kkrawczyk123** [comment]
I am not sure about purging process so I reported a new issue https://github.com/opendatakit/aggregate/issues/307. Confirm that other process works.
@opendatakit-bot unlabel "needs testing"

**Pull request merged**